tests/trainer/test_trainer.py

```diff
@@ -103,7 +103,8 @@ def test_no_val_end_module(monkeypatch, tmpdir, tmpdir_server, url_ckpt):
     model_2.eval()
 
 
-def test_gradient_accumulation_scheduling(tmpdir):
+@pytest.mark.parametrize('schedule_expected', [({1: 2, 3: 4}, [1, 2, 4]), (3, [3, 3, 3]), (4, [4, 4, 4])])
+def test_gradient_accumulation_scheduling(tmpdir, schedule_expected):
```

Comment:
you can split the two args like this for better readability

```suggestion
@pytest.mark.parametrize(['schedule', 'expected'], [
        pytest.param({1: 2, 3: 4}, [1, 2, 4]), 
        pytest.param(3, [3, 3, 3]), 
        pytest.param(4, [4, 4, 4])
    ]
)
```
def test_gradient_accumulation_scheduling(tmpdir, schedule, expected):
```

Reply:
Thanks. I wasn't sure how to do that with pytest. I'll make that change.